techmatters/flex-plugins
# CHI 1121 Accessibility Enhancements for Case List

Primary reviewer:@GPaoloni  

## Description
- Removed 'expand' button and added a new styled component `CLCaseIDButton` based on button element for CaseId - keyboard buttons, tab and enter/space work
- Styling for CaseId column has been modified according to this [mock](https://marvelapp.com/prototype/6gcabah/screen/86290860)
    - Note that the sort arrow button  is not implemented. This requires more investigation
- Tested that table content is accessible using screen readers. 
- Added outline on case rows when hovered on mouse
- Adjusted the table column widths to accommodate the new changes. However, I did not change the order to avoid conflicts with upcoming changes to the Epic- see [mock](https://marvelapp.com/prototype/6gcabah/screen/86290860)

### Next Steps:
- Table headers are styled based on MUI components and require a larger refactor be accessible with a screen reader and keyboard. Refer to James Scholes's report and email on sorting. Namely, buttons need to be used for the sortable table headers and th elements should replace MUI's table components.
- Pagination is accessible by keyboard but there is no outline at focus on tab. I will address this as a different PR given there are some major styling refactors.
- Sort arrow button is not implemented as shown in mocks. This requires more investigation or just waiting until the Flex UI update. 

### Checklist
- [n/a] New tests added
- [n/a] Tested for chat contacts
- [n/a] Tested for call contacts

### Related Issues
Fixes [#1121](https://bugs.benetech.org/browse/CHI-1121)

### Verification steps
![UI Changes to Case # column and other ](https://user-images.githubusercontent.com/42842977/165871108-7a60bfb2-04c0-4c33-82cf-2457d700ce8a.png)
- Use keyboard to navigate through case list page with tab and open a case with enter or space
- Use Screen Reader  to navigate through case list page and check that detailed information is also accessible
- Check with the updated [mocks](https://marvelapp.com/prototype/6gcabah/screen/86290860) to ensure implementation is identical.
